front/src/modules/trainschedule/components/ImportTimetableItem/ImportTimetableItemConfig.tsx

```diff
@@ -269,18 +291,189 @@ const ImportTimetableItemConfig = ({
 
       const trainSchedule: ImportedTrainSchedule = {
         trainNumber,
-        rollingStock: rollingStockViriato, // RollingStocks in viriato files rarely have the correct format
+        rollingStock: rollingStockXml, // RollingStocks in xml files rarely have the correct format
         departureTime: `${startDate} ${firstDepartureTimeformatted}`,
         arrivalTime: `${startDate} ${lastDepartureTimeformatted}`,
         departure: '', // Default for testing
         steps: adaptedSteps,
       };
-
+      trainSchedulesByTrainPartId[trainPartId] = trainSchedule;
       trainSchedules.push(trainSchedule);
     });
+
+    trainGroups.forEach((trainGroup) => {
+      const pacedTrainId = trainGroup.getAttribute('id')!;
+
+      const trainRefs = Array.from(trainGroup.getElementsByTagName('trainRef'));
+      pacedTrains[pacedTrainId] = trainRefs
+        .map((trainRef) => {
+          const trainId = trainRef.getAttribute('ref');
+          const trainElement = xmlDoc.querySelector(`train[id="${trainId}"]`);
+
+          const trainPartRef = trainElement?.querySelector('trainPartRef')?.getAttribute('ref');
+
+          return trainPartRef ? trainSchedulesByTrainPartId[trainPartRef] : undefined;
+        })
+        .filter((schedule): schedule is ImportedTrainSchedule => schedule !== undefined);
+    });
+
+    const getRelativeStepsTimeAndNames = (trainSchedule: ImportedTrainSchedule): string => {
+      const departureTime = new Date(trainSchedule.departureTime);
+
+      const stepsWithRelativeTimesAndNames = trainSchedule.steps.map((step) => {
+        const arrival = new Date(step.arrivalTime);
+        const departure = new Date(step.departureTime);
+
+        return {
+          name: step.name,
+          relativeArrivalSeconds: Math.round(
+            Duration.subtractDate(arrival, departureTime).total('second')
+          ),
+          relativeDepartureSeconds: Math.round(
+            Duration.subtractDate(departure, departureTime).total('second')
+          ),
+        };
+      });
+
+      return JSON.stringify(stepsWithRelativeTimesAndNames);
+    };
+
+    const findMostFrequentScheduleInPacedTrain = (schedules: ImportedTrainSchedule[]) => {
+      const scheduleOccurrences = new Map<
+        string,
+        { count: number; schedule: ImportedTrainSchedule }
+      >();
+
+      schedules.forEach((schedule) => {
+        const relativeSteps = getRelativeStepsTimeAndNames(schedule);
+        if (scheduleOccurrences.has(relativeSteps)) {
+          const entry = scheduleOccurrences.get(relativeSteps)!;
+          scheduleOccurrences.set(relativeSteps, {
+            ...entry,
+            count: entry.count + 1,
+          });
```

Comment:
Nit: `entry.count++` should be enough here.

Reply:
Sounds like this comment got lost?

Reply:
>  get an error from ESLint when I use the ++ instead of the +1 🤔

Ah, yeah (it's an annoying one, we should probably get rid of it). My main point is that we don't need to call `scheduleOccurrences.set()`. We should be able to replace that call with `entry.count += 1`.